docs/proposals/02-backupinfra.md

```diff
@@ -0,0 +1,232 @@
+  lastOperation: ...
+```
+
+### Workflow
+
+- Gardener administrator will configure the cloudProfile with backup infra credentials and provider config as follows.
+
+```yaml
+# CloudProfile.yaml:
+Spec:
+    backup:
+        provider: aws
+        region: eu-qest-1
+        secretRef:
+            name: backup-operator-aws
+            namespace: garden
+```
+Here CloudProfileController will interpret this spec as follows:
+- If `spec.backup.region` is not nil,
+  - then respect it, i.e. use the provider and unique region field mentioned there for backupContainer.
+  - Here Preferably, `spec.backup.region` field will be unique, since for cross provider, it doesn’t make much sense. Since region name will be different for different providers.
+- Otherwise, spec.backup.region is nil then,
+  - If same provider case i.e. spec.backup.provider = spec.(type of provider)  or nil then
+    - per region from `spec.aws.constraints.regions` create a `BackupContainer` instance. This can be done lazily if required i.e. create `BackupContainer` instance for region only if some seed actually spawned in the region is registered. This will avoid creating iaas bucket even if no seed is registered in that region, but region is listed in cloudprofile.
+    - Shoot controller will choose backup container as per the seed region. (With shoot control plane migration also, seed’s availibility zone might change but the region will be remaining same as per current scope)
+  - Otherwise cross provider case i.e. spec.backup.provider = spec.(type of provider)
+    - report validation error: Since we can’t expect `spec.backup.provider` = aws  to support region in, `spec.packet.constraint.region`.
+
+
+![sequence-diagram](Backup-Infrastructure-Provisioning-sequence-diagram.svg)
+
+#### Reconciliation
+
+Reconciliation on backup entry in seed cluster mostly comes in picture at the time of deletion. But we can add initialization steps like creation of [directory](#terminology) specific to shoot in backup bucket. We can simply create BackupEntry at the time of shoot deletion as well.
+
+#### Deletion
+- On shoot deletion, the BackupEntry instance i.e. shoot specific instance will get deletion timestamp because of ownerReference.
+- If `deletionGracePeriod` configured in GCM component configuration is expired, BackupInfrastructure Controller will delete the backup folder associated with it from backup object store.
+- Finally, it will remove the `finalizer` from backupEntry instance.
+
+### Alternative
+
+![sequence-diagram](Backup-Infrastructure-Provisioning-with-deletion-job.svg)
+
+## Discussion points / variations
+### Manual vs dynamic bucket creation
+- As per limit observed on different cloud providers, we can have single bucket for backups on one cloud providers. So, we could avoid the little complexity introduced in above approach by pre-provisioning buckets as a part of landscape setup. But there won't be anybody to detect bucket existence and its reconciliation.
```

Comment:
> as a part of landscape setup

This should be avoided (there are already a lot of configuration steps in the installation process, let's try not to extend it).

Reply:
Yes agreed. There are some points, i just wanted to mention in doc, that lists some other variants we had considered while designing. But wont be part of actual design.